control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -2953,6 +2953,15 @@ func (r *HostedControlPlaneReconciler) reconcileKubeAPIServer(ctx context.Contex
 			if hcp.Spec.SecretEncryption.KMS == nil {
 				return fmt.Errorf("kms metadata not specified")
 			}
+
+			// If an EncryptionConfig exists, it is passed down when generating
+			// the new EncryptionConfig to make sure that we keep the original
+			// KMS API version.
+			err := r.Get(ctx, client.ObjectKeyFromObject(encryptionConfigFile), encryptionConfigFile)
+			if err != nil && !apierrors.IsNotFound(err) {
+				return fmt.Errorf("failed to get encryption config secret: %w", err)
```

Comment:
this is not needed, `createOrUpdate()` does that automatically before calling the Reconcile function

Reply:
no, this is only to get the object key (namespaced name). when `createOrUpdate` is called, it does a get first and updates the obj pointer (similar to what you are doing here but implicitly) then calls the reconcile function.